CMakeLists.txt

```diff
@@ -52,6 +52,9 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}"
 add_library(${PROJECT_NAME} src/console.cpp)
 set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION
                ${CONSOLE_BRIDGE_MAJOR_VERSION}.${CONSOLE_BRIDGE_MINOR_VERSION})
+target_include_directories(${PROJECT_NAME} INTERFACE
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
```

Comment:
The `include_directories` call above should be removed and the binary for being added here. Then the letters should be changed from `INTERFACE` to `PUBLIC`.

Reply:
This doesn't work.

```
CMake Error in CMakeLists.txt:
  Target "console_bridge" INTERFACE_INCLUDE_DIRECTORIES property contains
  path:

    "/home/ubuntu/ros2_ws/build/console_bridge_vendor/console_bridge-0.5.0-prefix/src/console_bridge-0.5.0-build"

  which is prefixed in the build directory.
```

Reply:
Since you didn't share how you changed the CMake code I will just say I think it was changed incorrectly. Please either push it to this branch or share it somewhere else to allow commenting on the actual diff.

Reply:
the following is an incremental change to accompany the inclusion of `${CMAKE_CURRENT_SOURCE_DIR}/include` on this line

~~~
diff --git a/CMakeLists.txt b/CMakeLists.txt
index d136b4f..880a7ee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ if(MSVC OR MSVC90 OR MSVC10)
   set(MSVC ON)
 endif (MSVC OR MSVC90 OR MSVC10)
 
-include_directories(include ${PROJECT_BINARY_DIR})
+include_directories(${PROJECT_BINARY_DIR})
 
 if(NOT DEFINED BUILD_SHARED_LIBS)
   option(BUILD_SHARED_LIBS "Build dynamically-linked binaries" ON)
~~~

I think you're also suggesting to include the `${PROJECT_BINARY_DIR}` in this `target_include_directories` call?

Reply:
is this what you're suggesting @dirk-thomas ?

~~~
diff --git a/CMakeLists.txt b/CMakeLists.txt
index d136b4f..ec368cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,6 @@ if(MSVC OR MSVC90 OR MSVC10)
   set(MSVC ON)
 endif (MSVC OR MSVC90 OR MSVC10)
 
-include_directories(include ${PROJECT_BINARY_DIR})
 
 if(NOT DEFINED BUILD_SHARED_LIBS)
   option(BUILD_SHARED_LIBS "Build dynamically-linked binaries" ON)
@@ -53,6 +52,7 @@ add_library(${PROJECT_NAME} src/console.cpp)
 set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION
                ${CONSOLE_BRIDGE_MAJOR_VERSION}.${CONSOLE_BRIDGE_MINOR_VERSION})
 target_include_directories(${PROJECT_NAME} INTERFACE
+       $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
        $<INSTALL_INTERFACE:include>)
 generate_export_header(${PROJECT_NAME}
~~~

Reply:
Almost, I would keep the order of the two `BUILD_INTERFACE` the same as before though.

Reply:
5d54ec0